fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/lowcardinality/DecodeCollector.java

```diff
@@ -357,6 +361,43 @@ public DecodeInfo visitPhysicalHashAggregate(OptExpression optExpression, Decode
         return info;
     }
 
+    @Override
+    public DecodeInfo visitPhysicalTableFunction(OptExpression optExpression, DecodeInfo context) {
+        if (context.outputStringColumns.isEmpty()) {
+            return DecodeInfo.EMPTY;
+        }
+        DecodeInfo info = context.createOutputInfo();
+        PhysicalTableFunctionOperator tableFunc = optExpression.getOp().cast();
+
+        if (!FunctionSet.UNNEST.equalsIgnoreCase(tableFunc.getFn().getFunctionName().getFunction())) {
+            info.decodeStringColumns.union(info.inputStringColumns);
+            info.decodeStringColumns.intersect(tableFunc.getFnParamColumnRefs());
+            info.inputStringColumns.except(info.decodeStringColumns);
+        }
+
+        info.outputStringColumns.clear();
+        for (ColumnRefOperator outerColRef : tableFunc.getOuterColRefs()) {
+            if (info.inputStringColumns.contains(outerColRef)) {
+                info.outputStringColumns.union(outerColRef);
+            }
+        }
+
+        if (!FunctionSet.UNNEST.equalsIgnoreCase(tableFunc.getFn().getFunctionName().getFunction())) {
+            return info;
+        }
+
+        // unnest must be one input
+        ColumnRefOperator unnestOutput = tableFunc.getFnResultColRefs().get(0);
```

Comment:
unnest supports multiple parameters.  Do you want to only process one parameter scene?